Packs/Mattermost/Integrations/MattermostV2/MattermostV2.py

```diff
@@ -0,0 +1,1668 @@
+        channel_id = mirror.get('channel_id', '')
+        mirrors = json.loads(integration_context['mirrors'])
+        # Check for mirrors on the archived channel
+        channel_mirrors = list(filter(lambda m: channel_id == m['channel_id'], mirrors))
+        for mirror in channel_mirrors:
+            mirror['remove'] = True
+            demisto.mirrorInvestigation(mirror['investigation_id'], f'none:{mirror["mirror_direction"]}',
+                                        mirror['auto_close'])
+
+        set_to_integration_context_with_retries({'mirrors': mirrors}, OBJECTS_TO_KEYS)
+
+    return CommandResults(
+        readable_output=hr
+    )
+
+
+def list_users_command(client: HTTPClient, args: dict[str, Any]) -> CommandResults:
+    """ Lists users """
+    team_name = args.get('team_name', '')
+    channel_name = args.get('channel', '')
+    page = arg_to_number(args.get('page', DEFAULT_PAGE_NUMBER))
+    page_size = arg_to_number(args.get('page_size', DEFAULT_PAGE_SIZE))
+    limit = arg_to_number(args.get('limit', ''))
+
+    if limit:
+        page = DEFAULT_PAGE_NUMBER
+        page_size = limit
+
+    team_id = ''
+    if team_name:
+        team_details = client.get_team_request(team_name)
+        team_id = team_details.get('id', '')
+
+    channel_id = ''
+    if channel_name:
+        if not team_name:
+            raise DemistoException("Must provide a team name if a channel name was provided.")
+        channel_details = client.get_channel_by_name_and_team_name_request(team_name, channel_name)
+        channel_id = channel_details.get('id', '')
+        team_id = ''  # The search in Mattermost is done with an OR operator
+
+    params = {'page': page, 'per_page': page_size, 'in_team': team_id, 'in_channel': channel_id}
+    remove_nulls_from_dictionary(params)
+
+    users = client.list_users_request(params)
+
+    hr = tableToMarkdown('Users:', users, headers=['username', 'email', 'role', 'id'])
+    return CommandResults(
+        outputs_prefix='Mattermost.User',
+        outputs_key_field='id',
+        outputs=users,
+        readable_output=hr,
+    )
+
+
+def send_file_command(client: HTTPClient, args) -> CommandResults:
+    """ Sends a file """
+    channel_name = args.get('channel', '')
+    team_name = args.get('team_name', client.team_name)
+    message = args.get('message')
+    entry_id = args.get('entry_id') or args.get('file')
+    to = args.get('to', '')
+
+    demisto.debug(f'{to=}, {channel_name=}')
+
+    if (to and channel_name):
+        raise DemistoException("Cannot use both to and channel_name arguments")
+
+    if not to and not channel_name:
+        raise DemistoException("You must provide an to or channel_name arguments")
+
+    if to:
+        # create a new direct channel and send the message there
+        if re.match(emailRegex, to):
+            to = get_user_id_by_email(client, to)
+        else:
+            to = get_user_id_by_username(client, to)
```

Comment:
let's add debug logs here

Reply:
added, and added more debug logs in the code